packages/react/test/specs/components/Dropdown/Dropdown-test.tsx

```diff
@@ -156,11 +157,24 @@ describe('Dropdown', () => {
           open: true,
         }),
       )
+
+      wrapper
```

Comment:
actually, its not easy to see the reason for the event to happen this amount of times (which is requested by test). Could you, please, suggest why `onOpenChange` fires twice for each click?

Reply:
also, this check, apparently, verifies that on the subsequent open events `defaultHighlightedIndex` value is not honored. I would suggest to extract this check to the dedicated test case, with the corresponding name

Reply:
well, it is 3 because it opened a couple of times and closed once.

I just want to make sure that on opening it the second time, it opens and there is no item highlighted.

what do you think I should change?

Reply:
got it - I think that it would be more apparent if you would introduce a separate test case for that, the one that would follow a plan like:
- do whatever is necessary to prepare test state (arrange + act)
- verify conditions (assert)

Specifically to this example, it could be done the following way:
- introduce test case with a name like `is not set on second and subsequent openings`
- do the arrangement once there

```jsx
wrapper
  .find(...)
  . ...
  .simulate('click')
  .expect( .. ) // and this is where assertion phase starts
```

If there will be just single arrangement phase followed by assertion, it would be much easier to infer where the numbers come from :) currently the main problem is that there are several such phases within single test, so that the side effect of the previous arrangement affects the following one, which is not intuitive and harder to maintain

Reply:
split the test into two.